storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/UnblindedBlockKvStoreDatabase.java

```diff
@@ -100,6 +100,18 @@ public Optional<SignedBeaconBlock> getEarliestAvailableBlock() {
     return dao.getEarliestFinalizedBlock();
   }
 
+  @Override
+  public Optional<SignedBeaconBlock> getLastAvailableFinalizedBlock() {
+    return dao.getFinalizedCheckpoint()
+        .flatMap(
+            checkpoint -> getFinalizedBlock(checkpoint.toSlotAndBlockRoot(spec).getBlockRoot()));
+  }
+
+  @Override
+  public Optional<Bytes32> getFinalizedBlockRootBySlot(final UInt64 slot) {
+    return Optional.empty();
```

Comment:
Given how "normal" and potentially useful this method looks, I think its worth implementing it here. It would have to load the block and compute the hash tree root which isn't particularly efficient but better than suggesting the block doesn't exist.